gratipay/models/participant/__init__.py

```diff
@@ -81,6 +81,28 @@ def with_random_username(cls):
         return cls.from_username(username)
 
     @classmethod
+    def with_email_and_username(cls, email, username):
```

Comment:
I'd prefer to see this use the existing `username.safely_reserve` infrastructure for consistency and code reduction.